File: Array/Searching/Linear-Search/LinearSearch.c
#include <stdio.h>

int main() {
    int n, target;

    // Input size of array
    printf("Enter the size of the array: ");
    scanf("%d", &n);

    int arr[n];

    // Input elements of array
    printf("Enter the elements of the array:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Display original array
    printf("Original array: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    // Input target to search
    printf("Enter the element to search for: ");
    scanf("%d", &target);

    // Perform linear search
    int found = 0;
    for (int i = 0; i < n; i++) {
        if (arr[i] == target) {
            printf("Element %d found at index %d.\n", target, i);
            found = 1;
            break;
        }
    }

    if (!found) {
        printf("Element %d not found in the array.\n", target);
    }

    return 0;
}

// This program implements a linear search algorithm in C.

File: Array/Searching/Linear-Search/LinearSearch.java
import java.util.Scanner;

public class LinearSearch {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(System.in)) {

            // Input size of array
            System.out.print("Enter the size of the array: ");
            int n = sc.nextInt();
            int[] arr = new int[n];

            // Input elements of array
            System.out.println("Enter the elements of the array:");
            for (int i = 0; i < n; i++) {
                arr[i] = sc.nextInt();
            }

            // Display original array
            System.out.print("Original array: ");
            for (int num : arr) {
                System.out.print(num + " ");
            }
            System.out.println();

            // Input target to search
            System.out.print("Enter the element to search for: ");
            int target = sc.nextInt();

            // Perform linear search
            boolean found = false;
            for (int i = 0; i < n; i++) {
                if (arr[i] == target) {
                    System.out.println("Element " + target + " found at index " + i + ".");
                    found = true;
                    break;
                }
            }

            if (!found) {
                System.out.println("Element " + target + " not found in the array.");
            }
        }
    }
}

// This code implements a simple linear search algorithm in Java.